src/quality_result_gui/api/types/quality_error.py

```diff
@@ -88,6 +93,33 @@ def get_all_errors(self) -> List[QualityError]:
         errors = [errors_by_feature.errors for errors_by_feature in self.errors]
         return [item for sub_list in errors for item in sub_list]
 
+    def get_errors_filtered_by_error_ids(
+        self, error_ids: List[str]
+    ) -> List[QualityErrorsByFeature]:
+        errors: List[QualityErrorsByFeature] = []
+
+        for error in self.errors:
+            filtered_errors = error.get_errors_filtered_by_error_ids(error_ids)
+
+            if filtered_errors:
+                errors.append(
+                    QualityErrorsByFeature(
+                        error.feature_type,
+                        error.feature_id,
+                        filtered_errors,
+                    )
+                )
+
+        return errors
+
+    def get_index_for_feature_id(self, feature_id: str) -> Optional[int]:
+        index = [error for error in self.errors if error.feature_id == feature_id]
```

Comment:
maybe naming could be more clear, since this is not the index but the item to find instead?

or maybe switch to a sligthly more optimal approach with `return next(index for index, error in enumerate(self.errors) if error.feature_id == feature_id)`, this will only loop until the found index, as the current code loops the list once fully and once until the index that matches the item. probably not noticeable but unnecessary iteration still

Reply:
Refactored whole logic as model update started to slow down with 100+ errors. So these methods are not needed anymore although reason for slowness was error visualization (all errors were looped in every row insert/removal).